dts/arm/raspberrypi/rpi_pico/rp2350.dtsi

```diff
@@ -0,0 +1,357 @@
+/*
```

Comment:
It works SPI, I2C, ADC, and WDT.
And not worked DMA, PIO, TIMER should be removed.
Some modifications are needed to address these issues.

https://github.com/zephyrproject-rtos/zephyr/pull/77859/commits/11458a8b7a7b348d07ede4783b01d138a7757838

Reply:
The watchdog needs to be modified.
https://github.com/zephyrproject-rtos/zephyr/pull/77859/commits/97a59699c03902ed3bcfea3589fd6d44c37c6640

Reply:
Noted, thanks.

I happened to get the timers working yesterday, and that's pushed to the current `HEAD` of this branch. 
I'll look at your commits to enable the watchdog, SPI, and I2C and try to fold them in whilst retaining the correct author/signoff for any substantial bits of work you've done. If you're planning on tackling any other parts of the bringup, perhaps a good idea to coordinate either via the raspberrypi channel on Discord or the discussion on GitHub?

Reply:
I am looking at this from the perspective of whether there are any problems with adding the RP2350B. I fixed the watchdog issue because it happened to be easy, but I am not making any driver modifications, and if necessary, I will make a separate PR. (Therefore, I am doing a simple operation check to list the drivers that need to be modified.)

Discrod is also fine, but please bear with us as we may not be able to respond as quickly.